src/node_url.cc

```diff
@@ -1418,6 +1419,66 @@ namespace url {
                             v8::NewStringType::kNormal).ToLocalChecked());
   }
 
+  MaybeLocal<Value> URL::ToObject(Environment* env) {
+    Isolate* isolate = env->isolate();
+    Local<Context> context = env->context();
+    HandleScope handle_scope(isolate);
+    Context::Scope context_scope(context);
+
+    const Local<Value> undef = Undefined(isolate);
+    if (context_.flags & URL_FLAGS_FAILED)
+      return undef;
+
+    Local<Value> argv[9] = {
+      undef,
+      undef,
+      undef,
+      undef,
+      undef,
+      undef,
+      undef,
+      undef,
+      undef,
+    };
+    argv[ARG_FLAGS] = Integer::NewFromUnsigned(isolate, context_.flags);
+    if (context_.flags & URL_FLAGS_HAS_SCHEME)
+      argv[ARG_PROTOCOL] = OneByteString(isolate, context_.scheme.c_str());
+    if (context_.flags & URL_FLAGS_HAS_USERNAME)
+      argv[ARG_USERNAME] = UTF8STRING(isolate, context_.username);
+    if (context_.flags & URL_FLAGS_HAS_PASSWORD)
+      argv[ARG_PASSWORD] = UTF8STRING(isolate, context_.password);
+    if (context_.flags & URL_FLAGS_HAS_HOST)
+      argv[ARG_HOST] = UTF8STRING(isolate, context_.host);
+    if (context_.flags & URL_FLAGS_HAS_QUERY)
+      argv[ARG_QUERY] = UTF8STRING(isolate, context_.query);
+    if (context_.flags & URL_FLAGS_HAS_FRAGMENT)
+      argv[ARG_FRAGMENT] = UTF8STRING(isolate, context_.fragment);
+    if (context_.port > -1)
+      argv[ARG_PORT] = Integer::New(isolate, context_.port);
+    if (context_.flags & URL_FLAGS_HAS_PATH)
+      argv[ARG_PATH] = Copy(env, context_.path);
+
+    TryCatch try_catch(isolate);
+
+    MaybeLocal<Value> ret =
+        env->url_constructor_function()
+            ->Call(env->context(), undef, 9, argv);
+
+    if (ret.IsEmpty()) {
+      ClearFatalExceptionHandlers(env);
+      FatalException(isolate, try_catch);
+    }
+
+    return ret;
```

Comment:
I wasn't entirely sure what to do here, to be honest. I wanted to get it out so that @bmeck could play with it then tweak it from there. Your suggestion works for me tho :-)